### DISCUSSION TITLE
How to share it publicly, kinda like webserver, on win 10?

### DISCUSSION THREAD

**Discussion by paciox** [hlky/stable-diffusion-webui#426] [General]
Hello

I have a windows machine online.
I see this in prompt when I launch
Running on local URL:  http://localhost:7860/

To create a public link, set `share=True` in `launch()`.

How can I share it with other people online (friends)?

Does this starts a webserver reachable from outside?
If not, how to?

**Comment by altryne** [comment]
Yeah, this spits out a shareable *.gradio.app url that's specific to your machine. 
Don't change the python file though, add it as a param in relauncher.py with `--share` and don't forget to set a password for your friends an family with `--share-password PWD` 
Username will be `webui` and your password of choice

**Reply by paciox** [reply]
Managed to do it, but had to modify web.py

thanks

**Reply by Maergoth** [reply]
@paciox Can you share what you added, exactly? I couldn't figure out where to add it.

**Reply by altryne** [reply]
@Maergoth if you're using the webui.cmd then you have to edit the launch parameters in relauncher.py inside the scripts directory

**Reply by Maergoth** [reply]
```
import os, time

n = 0
while True:
    print('Relauncher: Launching...')
    if n > 0:
        print(f'\tRelaunch count: {n}')
    os.system("python scripts/webui.py")
    print('Relauncher: Process is ending. Relaunching in 0.5s...')
    n += 1
    time.sleep(0.5)
```

That's the entirety of my relauncher.py. I don't see where to add/edit anything obvious, but I'm probably just clueless.

**Reply by altryne** [reply]
`os.system("python scripts/webui.py --share")`

**Reply by Rapidery** [reply]
where do i find "relauncher.py" cause it doesnt exist in mine?

**Reply by jmtatsch** [reply]
Seems you should put these flags now in webui-user.sh

```
# Commandline arguments for webui.py, for example: export COMMANDLINE_ARGS="--medvram --opt-split-attention"
export COMMANDLINE_ARGS="--share"
```

**Reply by Nicoolodion** [reply]
> Seems you should put these flags now in webui-user.sh
> 
> ```
> # Commandline arguments for webui.py, for example: export COMMANDLINE_ARGS="--medvram --opt-split-attention"
> export COMMANDLINE_ARGS="--share"
> ```

Ok maybe I am a little dumb... Should I write "--share" in the commandline_arg in the webui-user.bat file? In the .sh file I don't find any "COMMANDLINE_ARGS"

**Comment by hifarrer** [comment]
Updated. Add it also on webui-user.bat

@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS="--share"

call webui.bat

**Reply by q447552640** [reply]
maybe you can you --listen after .bat or .sh like lanch.sh --listen

**Comment by JaosonMa** [comment]
#543
